src/ui/SetInputStep.js

```diff
@@ -51,21 +51,48 @@ function setInputStepInit() {
         video.onloadedmetadata = function(e) {
           video.play();
         };
+
+        document.getElementById('capture').addEventListener('click', function(stream){
+          context.drawImage(video, 0, 0, 400, 300);
+          options.onTakePhoto(canvas.toDataURL());
+        });
+
         document.getElementById('close').addEventListener('click', function () {
           stopStream(stream);
         });
       }
       function handleError(error) {
         console.log('navigator.getUserMedia error: ', error);
+
+        // when user dismissed the camera access (includes closing of prompt which requests for camera access)
+        if(error.message == 'Permission denied' || error.message == 'NotAllowedError' || error.message == 'PermissionDismissedError'){
+          document.getElementById('capture').addEventListener('click', function(e) {
+            alert('Enable camera access in order to take picture');
+          });
+        }
+        
+        // when user don't have webcam to use.
+        if(error.message == 'NotFoundError' || error.message == 'DevicesNotFoundError'){
+          alert('You do not have appropriate devices to use this Functionality');
+        }
+
+        // when webcam is already used by some other application
+        if(error.message == 'NotReadableError' || error.message == 'TrackStartError' || error.message == 'Concurrent mic process limit'){
+          alert('Your webcam is already in use by some other application');
+        }
+
+        // when some of the requested constraints can't be satisfied like high frame rate or high resolution
+        if(error.message == 'OverconstrainedError' || error.message == 'ConstraintNotSatisfiedError'){
+          console.log('Requested Constraints can not be satisfied ', error);
```

Comment:
Shouldn't there be an alert here too?

Reply:
But it's not related to browser side and also constrained is passed from the backend so it should be in console i guess. what's your call

Reply:
if you still want i will change it to `alert`

Reply:
If the user will not be able to see the image, shouldn't the user be notified? Or will the user be able to see the image still?

Reply:
1. Error related to the constraints will not affect the image untill we change them in backend, currently there are only 2 constraints -
```js
 {
   video:true
   audio:false,
 }
```
So this is not related to image anyway.

2. Yes user will still see the image

Reply:
The main cause of the extra processing is due to 📷 access denial and not bcos of some constraints so user should not be alerted for that.

Please correct me if i am wrong 😊

Reply:
@HarshKhandeparkar when i was adding this check i thought this will help in debbuging ( if the `promise` is rejected ).

User can still use their webcam to capture a image or get a alert for anything else.

Reply:
Ok, thank you!